src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1203,6 +1203,8 @@ impl Store {
             }
         });
 
+        let force_all_mutable =
+            mutable_paths.contains(&RelativePath::new(Component::CurDir).unwrap()) || force_mutable;
```

Comment:
Why is `.` special? Does this not apply to any other `mutate_paths` values?

Reply:
[This line](https://github.com/pantsbuild/pants/blob/e2699f4e894e90d485960ff6bf9ff2a858db2385/src/rust/engine/fs/store/src/lib.rs#L1265) checks if a path is mutable or not, but it starts with the children of the sandbox root directory. Because `mutable_paths` contains all ancestors of all mutable paths, it will actually always contain the sandbox root if a mutable path exists, so checking if the sandbox root is mutable before the loop would not work.  The existing solution works for every other valid path, though as I mentioned in my top-level comment on the PR, I'm not sure how negative paths work here.